compiler-core/src/type_/error.rs

```diff
@@ -200,7 +200,7 @@ pub enum Error {
         name: EcoString,
         module_name: EcoString,
         value_constructors: Vec<EcoString>,
-        type_with_same_name: bool,
+        context: ModuleValueErrorContext,
```

Comment:
Let's split this field into two and keep the previous `type_with_same_name`, rather than having the same bool in each.

```rust
pub enum ModuleValueUsageContext {
    UnqualifiedImport
    ModuleAccess
}
```